source/plugins/ruby/in_containerinventory.rb

```diff
@@ -82,6 +83,15 @@ def enumerate
                   if hostName.empty? && !containerRecord["Computer"].empty?
                     hostName = containerRecord["Computer"]
                   end
+                  if addonTokenAdapterImageTag.empty? && ExtensionUtils.isAADMSIAuthMode()
```

Comment:
aadonTokenAdapter is a local variable in enumerate which is being intialized to empty, every time, which means this will be evaluated for all containers in MSI mode every minute, until this container is found. This should be a global, that  should be set the first time, and then just use it as the container image for the sidecar wont change during our lifetime until we are redeployed/updated/restarted.